lib/ansible/modules/network/netconf/netconf_config.py

```diff
@@ -214,68 +244,82 @@ def main():
     else:
         module.fail_json(msg='Option src or xml must be provided')
 
-    try:
-        xml.dom.minidom.parseString(config_xml)
+    local_connection = module._socket_path is None
 
-    except Exception as e:
-        module.fail_json(msg='error parsing XML: %s' % to_native(e), exception=traceback.format_exc())
-
-    nckwargs = dict(
-        host=module.params['host'],
-        port=module.params['port'],
-        hostkey_verify=module.params['hostkey_verify'],
-        allow_agent=module.params['allow_agent'],
-        look_for_keys=module.params['look_for_keys'],
-        username=module.params['username'],
-        password=module.params['password'],
-    )
+    if not local_connection:
+        m = Connection(module._socket_path)
+        capabilities = module.from_json(m.get_capabilities())
```

Comment:
I have been hitting an issue here, where `m` ends up being a module_utils.connection.Connection without a `get_capabilities` method. I'm using `ansible_network_os=default` to access an openyuma server. That works without using `ansible_connection=netconf`